### ISSUE TITLE
Paraview Threshold

### ISSUE THREAD

**Issue by dzatelli** [vmtk/vmtk#259]
Hi! I'm new in this community. I have a question.. I'm using Threshold filter to extract surfaces one by one, but the filter seems to find 6 surfaces while there are only 4.  I've tried to remove them using the clip function but, as I extract the surface, the number doesn't change. Do you have any idea how to remove them?

**Comment by rlizzo** [comment]
Hi @dzatelli, Welcome to the community!

I'm a bit confused by your exact issue, in what capacity are you using VMTK?  Do you mean that you are experiencing issues when using the Paraview "threshold" and "clip" filters on a surface generated with VMTK? or are you using the VMTK `vmtkthresholds` and `vmtksurfaceclipper` operations? 

Also, when you say you want to "extract surfaces", what metric are you "selecting" based on (connected components, largest region, closests region to a certain point, etc.)? Do you have labels attached as cell data (ie. the definition for each triangle) or as point data (the x,y,z locations of the points) in the vtkPolyData file? (no worries if you don't know, that's a bit of a technical question)

If you are able to share more details it would be greatly appreciated. It would be most helpful if you provided the following info:

VMTK version
Python Version
Operating System
Exact copy/paste of pype script used which is giving issue
Screenshots of input data and resulting output
A description of the expected results and how the output differs. 

In addition, if you are able to share the input data file (it will not be made public), I can try to reproduce the issue on my end and potentially provide an outline of a direct method to the solution. you can upload it to a secure dropbox account at the following link: https://www.dropbox.com/request/m91bmqYguz3PRUlTGT4f

**Comment by dzatelli** [comment]
Hi Rick! Thank you for your reply. I used vmtklevelsetsegmentation to
reproduce a carotid. Then I've done all the passages to smooth the surface,
clip and extend it. Now I have to divide the 4 surfaces (wall, cca, eca,
ica) and after that create the mesh. I'm using Paraview for this, and the
filters are the following: Extract surface --> Generate surface normals -->
Connectivity --> Threshold. With the last filter, we can find in properties
"minimum" and "maximum" that should be 0 and 3, because there should be 4
surfaces. It happens that in one carotid I find a maximum of 12 surfaces,
and I don't know how to eliminate them. In some cases I used Meshmixer and
smooth the carotid where there were the supposed surfaces signaled in
Paraview. But in this case, there are all the wrong surfaces in a unique
point and I can't eliminate them. There's a way to select only the right
surfaces or I don't know what?
The versions of the software I'm using are the following: vmtk-1.3 and
Paraview 5.50-RC2.
Thank you,
Daniela

Il Mar 10 Apr 2018, 02:37 Rick Izzo <notifications@github.com> ha scritto:
